### ISSUE TITLE
@cdklabs/generative-ai-cdk-constructs: Unable to use Aurora Postgres pgVector for a knowledge base as described in the documentation

### ISSUE THREAD

**Issue by malikalimoekhamedov** [awslabs/generative-ai-cdk-constructs#695]
### Describe the bug

Currently, Aurora's pgVector cannot be used as a knowledge base source, as described [here](https://awslabs.github.io/generative-ai-cdk-constructs/src/cdk-lib/bedrock/).

I've seen @krokoko mentioning this limitation in his response [here](https://github.com/awslabs/generative-ai-cdk-constructs/pull/414#issuecomment-2108287110). However, I couldn't find the associated issue he said he'd open. So, I'm filing one here.

The process goes well until the custom resource needs to configure pgVector and has to communicate in Python with Postgres through psycopg. Once CDK deployment reaches this stage, it throws the following error:

```shell
Received response status [FAILED] from custom resource. Message returned: Error: Unable to import module 'custom_resources': No module named 'psycopg2._psycopg'

Logs: /aws/lambda/STACK-AmazonAuroraPgVectorCRProviderCustomResourcesF-blah

at invokeUserFunction (/var/task/framework.js:2:6)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async onEvent (/var/task/framework.js:1:369)
at async Runtime.handler (/var/task/cfn-response.js:1:1826) (RequestId: blah-blah)
```

### Expected Behavior

The deployment succeeds as documented.

### Current Behavior

The custom resource lambda function can't find the psycopg2 library.

```shell
Received response status [FAILED] from custom resource. Message returned: Error: Unable to import module 'custom_resources': No module named 'psycopg2._psycopg'

Logs: /aws/lambda/STACK-AmazonAuroraPgVectorCRProviderCustomResourcesF-blah

at invokeUserFunction (/var/task/framework.js:2:6)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async onEvent (/var/task/framework.js:1:369)
at async Runtime.handler (/var/task/cfn-response.js:1:1826) (RequestId: blah-blah)
```

### Reproduction Steps

```typescript
const vectorStoreAurora = new amazonaurora.AmazonAuroraDefaultVectorStore(
  this,
  'AuroraDefaultVectorStore',
  {
    embeddingsModelVectorDimension:
      bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V2_1024.vectorDimensions!,
  }
);

this.ref = new bedrock.KnowledgeBase(this, 'KnowledgeBaseAurora', {
  vectorStore: vectorStoreAurora,
  embeddingsModel: bedrock.BedrockFoundationModel.TITAN_EMBED_TEXT_V2_1024,
  instruction: 'Some instructions.',
  description: 'Some description.',
});
```

### Possible Solution

The Internet says it could be an issue of x86 vs. arm64 incompatibility, where the custom resource lambda's architecture would be incompatible with the psycopg2-binary package.

@krokoko says he only experiences it on MacOS.x86 vs. arm64 incompatibility issue

### CDK CLI Version

2.158.0 (build 4b8714d)

### Framework Version

0.1.264

### Node.js Version

v22.8.0

### OS

MacOS Sonoma 14.6.1

### Language

Typescript

### Language Version

5.6.2

### Region experiencing the issue

us-west-2

### Code modification

No

### Service quota

- [X] I have reviewed the service quotas for this construct

**Comment by krokoko** [comment]
Hi @malikalimoekhamedov ! Thanks for reporting this issue. It seems indeed to be related to the specific platform. I will investigate the issue

**Comment by maxtybar** [comment]
Looking into fixing this as well.

**Comment by krokoko** [comment]
Closing as fix in #724
